pkg/bisect/bisect.go

```diff
@@ -577,33 +585,39 @@ func (env *env) test() (*testResult, error) {
 	results, err := env.inst.Test(numTests, cfg.Repro.Syz, cfg.Repro.Opts, cfg.Repro.C)
 	env.testTime += time.Since(testStart)
 	if err != nil {
-		res.rep = &report.Report{
-			Title: fmt.Sprintf("failed testing reproducer on %v: %v", current.Hash, err),
-		}
-		env.log(res.rep.Title)
-		return res, nil
+		problem := fmt.Sprintf("repro testing failure: %v", err)
+		env.log(problem)
+		return res, &InfraError{Title: problem}
 	}
-	bad, good, rep := env.processResults(current, results)
+	bad, good, infra, rep := env.processResults(current, results)
 	res.rep = rep
 	res.verdict = vcs.BisectSkip
+	if infra > len(results)/2 {
+		// More than 1/2 of runs failed with infrastructure error,
+		// no sense in continuing the bisection at the moment.
+		return res, &InfraError{Title: "More than 50% of runs failed with an infra error"}
+	}
 	if bad != 0 {
 		res.verdict = vcs.BisectBad
 		if !env.flaky && bad < good {
 			env.log("reproducer seems to be flaky")
 			env.flaky = true
 		}
-	} else if len(results)-good-bad > len(results)/3*2 {
-		// More than 2/3 of instances failed with infrastructure error,
-		// can't reliably tell that the commit is good.
-		res.verdict = vcs.BisectSkip
 	} else if good != 0 {
 		res.verdict = vcs.BisectGood
+	} else {
+		res.rep = &report.Report{
```

Comment:
What is err here? Isn't it handled above already?

Reply:
Indeed. Removed the mention of `err`.